pkg/resources/helpers.go

```diff
@@ -61,3 +61,41 @@ func setBoolProperty(d *schema.ResourceData, key string, property *sdk.BoolPrope
 	}
 	return nil
 }
+
+func getTagObjectIdentifier(v map[string]any) sdk.ObjectIdentifier {
+	if _, ok := v["database"]; ok {
+		if _, ok := v["schema"]; ok {
+			return sdk.NewSchemaObjectIdentifier(v["database"].(string), v["schema"].(string), v["name"].(string))
+		}
+		return sdk.NewDatabaseObjectIdentifier(v["database"].(string), v["name"].(string))
+	}
+	return sdk.NewAccountObjectIdentifier(v["name"].(string))
+}
+
+func getPropertyTags(d *schema.ResourceData, key string) []sdk.TagAssociation {
```

Comment:
What was the plan for tags and tags associations @scottwinkler? I can see that now they are supported in two places: for some resource, like schema here, they have `"tag": tagReferenceSchema` specified, but `tagReferenceSchema` has `Deprecated:  "Use the 'snowflake_tag_association' resource instead."`.

So I believe that for now we want to support both options, and with V1 we cut out this deprecated option, right?

Reply:
well i think it is a similar problem to the snowflake_parameters vs having the same attribute be on a resource. We also have to weigh the downside of basically doubling the number of resources, if customers want to tag every resource. i didn't like the implementation of tag blocks previously, because it was difficult to know when the order changed, and also it made for really long configuration code, but I could see maybe adding them back later with a slightly different syntax, e.g.:

```
snowflake_database "d" {
....
tags = [
{
tag_name = ""
tag_value = ""
},
{
tag_name = ""
tag_value = ""
},
...
]
}
```

That being said, if we add tags back in it would be problematic for people who use the existing tag association resource. They would need to set `ignore_changes = ["tags"] ` on every resource which uses tags so as not dissociate every tag they had set. https://developer.hashicorp.com/terraform/language/meta-arguments/lifecycle#ignore_changes. At the time it was too complicated to have a tags block on every resource, which is why it was made into its own resource. Now that we have a more stable API, I'm not sure its beneficial anymore.